047-TrafficControlWithDapr/Coach/Solution-08.md

```diff
@@ -15,166 +15,77 @@ To complete this challenge, you must reach the following goals:
     ```
 
-1. 	Navigate to the `Resources/TrafficControlService` directory & use the Azure Container Registry task to build your image from source.
+1.  Navigate to the `Resources/TrafficControlService` directory & use the Azure Container Registry task to build your image from source.
 
     ```shell
     az acr build --registry <container-registry-name> --image trafficcontrolservice:assignment08 .
     ```
 
-1. 	Navigate to the `Resources/FineCollectionService` directory & use the Azure Container Registry task to build your image from source.
-  
+1.  Navigate to the `Resources/FineCollectionService` directory & use the Azure Container Registry task to build your image from source.
+
     ```shell
-    az acr build --registry <container-registry-name> --image trafficcontrolservice:assignment08 .		
+    az acr build --registry <container-registry-name> --image trafficcontrolservice:assignment08 .
     ```
 
 ### Step 4: Deploy container images to Azure Kubernetes Service
 
 Now that your container images have been uploaded to the Azure Container Registry, you can deploy these images to your Azure Kubernetes Service. Deployment spec files have been added to each service to make this easier. You will need to customize them to reference your container registry path & AKS ingress.
 
-1.	Open the `Resources/dapr/components/fine-collection-service.yaml` file and update the container registry name to be the one you have deployed.
-
-    ```yaml
-    spec:
-      containers:
-      - name: finecollectionservice
-        image: <container-registry-name>.azurecr.io/finecollectionservice:assignment08
-    ```
-
-1.	Modify the ingress host to match your AKS instance's HTTP application routing domain. You can query for this if you don't have it.
-
-    ```shell
-    az aks show --resource-group <resource-group-name> --name <aks-name> --query="addonProfiles.httpApplicationRouting.config.HTTPApplicationRoutingZoneName"
-    ```
-
-    ```shell
-    "e13e6fb6d2534a41ae60.southcentralus.aksapp.io"
-    ```
-
-    ```yaml
-    spec:
-    rules:
-    - host: finecollectionservice.<aks-http-application-routing-zone-name>
-    ```
-    
-1.  Repeat these steps for the `TrafficControlService` and the `VehicleRegistrationService`.
-
-1.  Grant your AKS instance access to pull images from your Azure Container Registry.
+_IMPORTANT: The Azure Container Registry has the **admin** account enabled to make this demo easier to deploy (doesn't require the deployer to have Owner access to the subscription or resource group the Azure resources are deployed to). **This is not a best practice!** In a production deployment, use a managed identity or service principal to authenticate between the AKS cluster & the ACR. See the [documentation](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-authentication?tabs=azure-cli) for more about the options and how to set up._
```

Comment:
I this a good idea to implement a bad practice, even to remove the need for "owner" permission to deploy the hack's Azure environment?

Reply:
I don't know, no customer I have is granted Owner access (not even to their resource group), I was trying to find ways to make this easier to use, what do you think?

Reply:
I guess this is okay to leave in.  But definitely need to emphasize this is an anti-pattern when delivering.

Reply:
no doubt